### PR TITLE
Apply compose hooks to all snapshot calls

### PR DESCRIPTION
**Pull request by gamepro65**
Add support for ViewGroups which have compose components since AbsComposeView can be added anywhere. If we detect compose on the classpath then apply our hooks instead of only applying to the Composable snapshot.

### CODE REVIEW

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt [hunk 1]
```diff
@@ -622,6 +622,15 @@ class Paparazzi @JvmOverloads constructor(
     private val isVerifying: Boolean =
       System.getProperty("paparazzi.test.verify")?.toBoolean() == true
 
+    private val hasComposeRuntime: Boolean =
```
Comment (TWiStErRob) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:625]: How often does this change? Should it be cached either `by lazy` or in `companion object`?

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt [hunk 2]
```diff
@@ -193,20 +191,11 @@ class Paparazzi @JvmOverloads constructor(
...
-    PaparazziComposeOwner.register(parent)
-    hostView.setContent(composable)
-
-    try {
-      snapshot(parent, name)
-    } finally {
-      forceReleaseComposeReferenceLeaks()
+    val hostView = ComposeView(context).apply {
+      setContent(composable)
     }
```
Comment (jrodbx) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:196]:
```suggestion
    val hostView = ComposeView(context)
    hostView.setContent(composable)
```

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt [hunk 3]
```diff
@@ -292,6 +281,14 @@ class Paparazzi @JvmOverloads constructor(
           // Initialize the choreographer at time=0.
         }
 
+        if (hasComposeRuntime) {
+          // AbstractComposeView will attempt to resolve its parent's CompositionContext, which
```
Comment (jrodbx) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:285]:
```suggestion
          // // During onAttachedToWindow, AbstractComposeView will attempt to resolve its parent's CompositionContext, which
```

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt (hunk 1 above)
Comment (jrodbx) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:625]: Will be calculated for each new Paparazzi rule creation for each test method run.  Fine for now, unless it's quick to profile and improve via companion object.

**Review by jrodbx — APPROVED** [review]
Clever!

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt (hunk 1 above)
Comment (jrodbx) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:625]: Oh, this is already in `companion object`.  Never mind!

Comment (TWiStErRob) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:634]: Oops 🤐, need to expand context on GH 😞

**Pull request merged**